hack/tools/go.mod

```diff
@@ -10,6 +10,7 @@ require (
 	github.com/jcchavezs/porto v0.6.0
 	go.uber.org/automaxprocs v1.5.2
 	go.uber.org/mock v0.4.0
+	golang.org/x/exp v0.0.0-20240416160154-fe59bbe5cc7f
```

Comment:
@liggitt: You were concerned about adding a dependency here.

Perhaps it would be better to always install the latest apidiff? Then we don't need to bother with updating this dependency.

There's no guarantee that the latest release will work, but that IMHO is okay for this script (optional, not required to work).

Reply:
we don't even have to install latest that way, right? we could bake a specific version into the install command. I think we did that with zeitgeist as well for the same reason in https://github.com/kubernetes/kubernetes/pull/118076/files

Reply:
apidiff doesn't have tagged versions.

Reply:
can't we use a pseudoversion to install?

Reply:
I suppose. But do we want to to use a fixed revision instead of always using "the latest"?

It will be another, non-obvious thing that needs to be bumped occasionally, probably when someone notices a problem. This is bound to happen at some point, whereas always using the latest might never cause problems. In klog, we've always used the latest without problems in the last five years.

OTOH, there is a certain (low?) risk that a future revision does something nasty in our build environment.

:shrug:

Reply:
> ... in our build environment.

Or on a developer's machine.

Reply:
I've pushed a version which uses `go install @latest`

Reply:
if this is advisory and non-blocking and won't be an issue in release branches, I can live with `@latest`. If this was blocking and would get branched into release branches and live there, I'd want to pin.

Reply:
Check:
- [x] not blocking
- [x] not branched

Can you LGTM or do you want to leave that to someone else?

/hold

To prevent accidental merge (is self-approved).